subxt/src/book/usage/blocks.rs

```diff
@@ -27,13 +27,59 @@
 //!
-//! ## Example
+//! ## Decoding Extrinsics
 //!
-//! Given a block, you can [download the block body](crate::blocks::Block::extrinsics()) and iterate over
-//! the extrinsics stored within it. From there, you can decode the extrinsics and access various details,
-//! including the associated events:
+//! Given a block, you can [download the block body](crate::blocks::Block::extrinsics()) and [iterate over
+//! the extrinsics](crate::blocks::Extrinsics::iter()) stored within it. The extrinsics yielded, are of type
+//! [ExtrinsicDetails](crate::blocks::ExtrinsicDetails), which is just a blob of bytes that also stores which
+//! pallet and call in that pallet it belongs to. It also contains information about signed extensions that
+//! have been used for submitting this extrinsic.
+//!
+//! To use the extrinsic, you probably want to decode it into a concrete Rust type. These Rust types representing
+//! extrinsics from different pallets can be generated from metadata using the subxt macro or the CLI tool.
+//!
+//! When decoding the extrinsic into a static type you have two options:
+//!
+//! ### Statically decode the extrinsics into [the root extrinsic type](crate::blocks::ExtrinsicDetails::as_root_extrinsic())
+//!
+//! The root extrinsic type generated by subxt is a Rust enum with one variant for each pallet. Each of these
+//! variants has a field that is another enum whose variants cover all calls of the respective pallet.
+//! If the extrinsic bytes are valid and your metadata matches the chains metadata, decoding the bytes of an extrinsic into
+//! this root extrinsic type should always succeed.
+//! This example shows how to subscribe to blocks, and decode the extrinsics in each block into the root extrinsic type.
+//! Once decoded, we can access details about the extrinsic, including the associated events and signed extensions.
 //!
 //! ```rust,ignore
 #![doc = include_str!("../../../examples/blocks_subscribing.rs")]
 //! ```
 //!
+//! ### Statically decode the extrinsic into [a specific pallet call](crate::blocks::ExtrinsicDetails::as_extrinsic())
+//!
+//! . This is
+//! useful if you are expecting a specific extrinsic to be part of some block. If the extrinsic you try to decode
+//! is a different extrinsic, an `Ok(None)` value is returned from [`as_extrinsic::<T>()`](crate::blocks::ExtrinsicDetails::as_extrinsic());
+//!
+//! If you are only interested in a specific kind of extrinsic in a block you can also [iterate over all of them](crate::blocks::Extrinsics::find),
+//! get only [the first one](crate::blocks::Extrinsics::find_first), or [the last one](crate::blocks::Extrinsics::find_last)
+//! given some the [Extrinsics](crate::blocks::Extrinsics) from a blocks body.
+//!
+//! The following example monitors `TransferKeepAlive` extrinsics on the polkadot network.
+//! We statically decode them and also make use of signed
+//!
+//! ```rust,ignore  
+#![doc = include_str!("../../../examples/polkadot_monitor.rs")]
```

Comment:
For consistency could we prefix the example name with `blocks_`; perhaps `blocks_static_decoding.rs` since that's the key thing that we are showing